geopandas/tests/test_plotting.py

```diff
@@ -86,6 +86,26 @@ def test_default_colors(self):
         expected_colors = cmap(np.arange(self.N) / (self.N - 1))
         _check_colors(self.N, ax.collections[0].get_facecolors(), expected_colors)
 
+    def test_color_reindex(self):
```

Comment:
For completeness, could you add a test of the other example @martinfleis showed in #1634 for plotting?
```
# this is correct, since gdf.index and colors.index are in the same order
colors = pd.Series(['a', 'b', 'c', 'a', 'b', 'c', 'a', 'b', 'c', 'a'])
df.plot(colors, legend=True)
```

Reply:
Added a new test...I think I've covered all of the examples in #1634 now. Let me know if you want anything more, tweaks, etc!